### PR TITLE
fix(genai,#12152): PT_11b md#19 — section 7 unique, prose alignée sur la config GRPO exécutée

### LINKED ISSUE #12152: PT_11b : md#19 enseigne 5 hyperparametres GRPO contredits par la config executee (dont beta=0.04 qui crashe)
`PT_11b_multiseed_qwen35_4x100.ipynb` enseigne, en md#19, **cinq hyperparamètres
GRPO qui contredisent la configuration réellement exécutée** par la cellule de code
d'à côté — et l'un d'eux est précisément la valeur que le code évite parce qu'elle
**crashe**.

Vérifié firsthand sur `main` (38 cellules, md#19 et src#20 adjacentes).

## La contradiction

| paramètre | md#19 §1 | md#19 §2 | **src#20 exécuté** |
|---|---|---|---|
| `num_generations` | 4 | 4 | **8** |
| `beta` | 0.04 | 0.04 | **0.0** |
| `learning_rate` | 5e-6 | **1e-5** | 5e-6 |
| `per_device_train_batch_size` | *(batch=1 évoqué)* | 2 | **8** |
| `gradient_accumulation_steps` | — | 4 → « effective batch = 8 » | 4 → **batch effectif 32** |

`GRPO_CONFIG_DICT` porte ses propres justifications en commentaire, et elles
disent l'inverse de la prose :

```python
"num_generations": 8,              # #10603: defaut TRL 8 (vs 2)
"beta": 0.0,                       # DAPO (beta>0 = crash peft#3340 sur trl1.9.2/tf5.x)
"per_device_train_batch_size": 8,  # DOIT etre divisible par num_generations (8 % 8 == 0)
"gradient_accumulation_steps": 4,  # batch effectif 32 = 4 prompts x 8 generations/step
"learning_rate": 5e-6,
```

## Pourquoi ce n'est pas un simple drift de doc

1. **`beta = 0.04` est la valeur qui crashe.** Le commentaire in-source dit
   `beta>0 = crash peft#3340 sur trl1.9.2/tf5.x`. Un étudiant qui applique la
   prose — « coefficient KL (garde-fou contre drift) » — reproduit exactement le
   crash que la config committée contourne. La prose enseigne le piège.
2. **`per_device_train_batch_size = 2` viole l'invariant écrit à côté.**
   `2 % num_generations` ≠ 0 pour `num_generations = 8` ; la contrainte de
   divisibilité est explicitement commentée dans le code.
3. **`effective batch = 8` est faux d'un facteur 4** (32 réellement).
4. **`learning_rate` se contredit à l'intérieur du même markdown** : 5e-6 dans la
   première section, 1e-5 dans la seconde.

## Défaut structurel dans la même cellule

md#19 contient **deux sections concurrentes portant le même numéro** :

```
## 7. Configuration GRPO pour RLVR sur Qwen3.5-0.8B     <- base
...
### 7. Configuration GRPO pour RLVR sur Qwen3.5-0.8B    <- ajoutee
```

Un `##` suivi d'un `###` homonyme n'est pas une sous-section : c'est un
dédoublement. C'est la pathologie de hiérarchie markdown que le user a signalée
comme classe (« diagrammes ascii, tailles de police d'indices, tableaux cassés…
le CI et moi avons tout laissé passer »), et elle est ici **la cause** de la
contradiction : la seconde rédaction ne remplace pas la première, elle s'ajoute à
côté.

## Correctif attendu

- **Une seule** section 7, la prose alignée sur `GRPO_CONFIG_DICT` — pas l'inverse :
  la config exécutée fait foi, la prose la décrit.
- Conserver la **pédagogie** des justifications (pourquoi `beta = 0` sur cette
  stack, pourquoi la divisibilité `batch % num_generations == 0`, pourquoi le
  batch effectif est 32) : c'est du contenu de valeur, il manque juste d'être
  exact. Le commentaire in-source cite déjà #10603 et peft#3340 — la prose peut
  s'appuyer dessus.
- Ré-exécution non requise si seul le markdown bouge (C.2), mais **vérifier que
  les valeurs citées viennent bien de l'output de src#20**, qui echo le dict.

## Provenance

Trouvé par l'organe B.0 en re-scannant les 80 dernières PR mergées (#12143) : la
review Hermes sur #12059 portait ce constat en 🟡, **la PR a été mergée sans
réponse écrite**, et le défaut vit depuis sur `main`. C'est le cas fondateur du
correctif de garde — mais le défaut pédagogique, lui, se répare ici, dans un grain
dédié, pas dans l'organe.

Grain suggéré : **MED/notebook-python** — famille GenAI/PostTraining.

### PR DESCRIPTION
**Pull request by jsboige**
Grain: MED/notebook-python — lane myia-po-2023:CoursIA — prev: MED/notebook-lean #11721 (PR #12164)

## Perimetre (declare en debut de body)

**1 fichier** : `PT_11b_multiseed_qwen35_4x100.ipynb`, **cellule md#19 uniquement** (source markdown). Aucune cellule code touchee — outputs et `execution_count` intacts (exception C.2, modifs uniquement markdown).

## Le defaut (verifie firsthand sur worktree frais origin/main)

md#19 portait **deux sections concurrentes de numero 7** (`## 7.` base + `### 7.` ajoutee), dont la prose contredisait le `GRPO_CONFIG_DICT` **reellement execute** par src#20 — dont une valeur que le code evite parce qu'elle **crashe** :

| parametre | ancienne prose (les 2 sections) | **src#20 execute** |
|---|---|---|
| `num_generations` | 4 | **8** (#10603, defaut TRL) |
| `beta` | 0.04 « KL penalty Deepseek-R1 » | **0.0** (DAPO ; `beta>0` = crash peft#3340 sur trl1.9.2/tf5.x) |
| `learning_rate` | 5e-6 / 1e-5 (contradiction interne) | 5e-6 |
| `per_device_train_batch_size` | 2 (viole `batch % num_generations == 0`) | **8** |
| `batch effectif` | « 8 » | **32 = 4 prompts x 8 generations** |
| `max_completion_length` | 384 | **96** |

## Correctif

- **Une seule** section 7. La prose decrit la config executee (la config fait foi, pas l'inverse) ; chaque valeur citee est verifiee contre l'**output reel de src#20** (le dict echo : `num_generations = 8`, `beta = 0.0`, `learning_rate = 5e-06`, `max_completion_length = 96`, etc.).
- **Pedagogie conservee et recallee** : pourquoi `beta = 0` sur cette stack (crash peft#3340, garde-fou = le verificateur RLVR lui-meme), pourquoi la divisibilite `8 % 8 == 0` (groupes complets par micro-batch), pourquoi le batch effectif est le **produit** 8 x 4 = 32 et comment il produit les **400 groupes** que l'output du training compte (`Groupes informatifs : 138/400 = 34.5% (#10603)` — le tie-in est mesure, pas affirme). References #10603, peft#3340, #10289, DeepSeekMath arXiv:2402.03300 conservees.
- Bloc « Pourquoi GRPO plutot que PPO » (non numerote, sans contradiction) conserve verbatim.

## Validation

- Markdown-only : aucune re-execution requise (C.2), outputs/execution_count intacts (verifie : 38 cellules, 17 code, toutes exec non-null avec outputs).
- nbformat VALID ; `scan_md_hierarchy.py` : **0 flag sur la cellule 19** (avant : double numerotation) ; les flags HEADING-IN-LIST des cellules 7/13 sont pre-existants sur main (classe #12165, hors scope).

## Residuels signales (principe 3, non couverts ici)

- **Cells 0/35/36** : prose de l'ere G=2 (acceptance intro « num_generations=2 », « batch effectif 4 », colonne « Local » du tableau cell 36 `num_generations=2`) contredite par l'artefact execute G=8 — meme veine claims-vs-outputs, grain separe a ouvrir.
- L'output du training montre « SEED 42 (1/1) » : la derniere execution reelle couvre 1 seed sur les 4 revendiques par le titre — a arbitrer ailleurs (re-exec multi-seed ou recalage du recit).

Closes #12152

### CODE REVIEW

**Review by jsboige — COMMENTED** [review]
[Hermes] Vérifié sur 40b7711c par extraction du notebook au head SHA : les 8 valeurs de la prose (G=8, beta=0.0, batch=8, grad_accum=4, lr=5e-6, max_completion=96, max_steps=100, bf16+grad_ckpt) matchent exactement le `GRPO_CONFIG_DICT` de la cellule 20. Heading '## 7.' unique dans le notebook (cell 19) ✓. Cohérence arithmétique OK (batch effectif 32 = 4×8, 400 groupes = 4×100 steps). Le point beta=0 (KL supprimée, garde-fou = vérificateur de reward) est bien documenté avec la référence peft#3340.

**Comment by myia-ai-01** [comment]
Merge. Verifie firsthand plutot que sur le libelle « markdown-only » :

- `git diff origin/main...HEAD` : **0** ligne `"execution_count"` modifiee, **0** ligne `"output_type"` modifiee. La preuve d'execution est intacte — c'est bien un remede de prose, pas un notebook re-ecrit.
- 1 fichier, +35/-29.
- Cap : `not LIGHT (effective MED)` — pas de plafond en cause.

L'advisory « ASCII flowchart introductions » rouge est **mon** baseline perime (`pytest test_detect_ascii_flowchart.py`, 11 attendus pour 6 reels), corrige par #12166. Verifie sur le log du run : aucune introduction ASCII ici.
